File: src/main/java/no.nav.sbl/rest/domain/DecoratorDomain.java
package no.nav.sbl.rest.domain;

import java.util.List;

public class DecoratorDomain {
    public static class DecoratorConfig {
        public final String ident, navn, fornavn, etternavn;
        public final List<Enhet> enheter;

        public DecoratorConfig(Saksbehandler saksbehandler, List<Enhet> enheter) {
            this.ident = saksbehandler.ident;
            this.fornavn = saksbehandler.fornavn;
            this.etternavn = saksbehandler.etternavn;
            this.navn = saksbehandler.navn;
            this.enheter = enheter;
        }
    }

    public static class Enhet {
        public final String enhetId, navn;

        public Enhet(String enhetId, String navn) {
            this.enhetId = enhetId;
            this.navn = navn;
        }
    }

    public static class Saksbehandler {
        public final String ident, navn, fornavn, etternavn;

        public Saksbehandler(String ident, String fornavn, String etternavn) {
            this.ident = ident;
            this.fornavn = fornavn;
            this.etternavn = etternavn;
            this.navn = fornavn + " " + etternavn;
        }
    }
}

File: src/main/kotlin/no/nav/modiacontextholder/rest/model/DecoratorDomain.kt
package no.nav.modiacontextholder.rest.model

import kotlinx.serialization.Serializable

class DecoratorDomain {
    @Serializable
    data class DecoratorConfig(
        private val saksbehandler: Saksbehandler,
        val enheter: List<Enhet>,
    ) {
        val ident: String = saksbehandler.ident
        val navn: String = saksbehandler.navn
        val fornavn: String = saksbehandler.fornavn
        val etternavn: String = saksbehandler.etternavn
    }

    @Serializable
    data class Enhet(
        val enhetId: String,
        val navn: String,
        val type: String? = null,
        /**
         * GruppeId er id-en til enheten in Entra ID.
         * Det ble forespurt i august 2025 om å eksponere den i APIet vårt slik at folk
         * kan bruke verdien videre til kall til Entra via MS Graph
         */
        val gruppeId: String? = null,
        /**
         * Om enheten kan behandle oppgaver. `null` betyr at NORG2 ikke oppga verdien,
         * og likestilles med `false`.
         */
        val oppgavebehandler: Boolean? = null,
    )

    @Serializable
    data class Saksbehandler(
        val ident: String,
        val fornavn: String,
        val etternavn: String,
    ) {
        val navn: String = "$fornavn $etternavn"
    }

    @Serializable
    data class FnrAktorId(
        val fnr: String,
        val aktorId: String,
    )
}
